### ISSUE TITLE
Is there some reason why special characters are escaped (backslashed) in Pretty* / Vertical formats?

### ISSUE THREAD

**Issue by filimonov** [yandex/ClickHouse#1729]
Special characters currently are always escaped (backslashed) in human readable formats (Pretty* / Vertical). That behavior doen't seems natural and is controversial.

Some samples:
```sql
:) select arrayStringConcat(['a','b','c'],'\n') as test;
┌─test────┐
│ a\nb\nc │
└─────────┘

1 rows in set. Elapsed: 0.002 sec. 

:) select arrayStringConcat(['a','b','c'],'\n') as test\G
Row 1:
──────
test: a\nb\nc

1 rows in set. Elapsed: 0.001 sec. 

-- same with tabs (which are )
SELECT arrayStringConcat(['a', 'b', 'c'], '\t')

┌─arrayStringConcat([\'a\', \'b\', \'c\'], \'\\t\')─┐
│ a\tb\tc                                           │
└───────────────────────────────────────────────────┘

-- it looks like TSVRaw is the only printable format which doesn't do that
:) select arrayStringConcat(['a','b','c'],'\n') as test FORMAT TSVRaw;
a
b
c

1 rows in set. Elapsed: 0.003 sec. 

-- even in table headers
SELECT concat('a', 'b', 'c')

┌─concat(\'a\', \'b\', \'c\')─┐
│ abc                         │
└─────────────────────────────┘

'-- quite often do that:
:) create table backslash_test(status Enum8('VALID' = 1, 'DELETED' = 2) DEFAULT 'VALID') Engine=Memory;
:) show create table backslash_test;

SHOW CREATE TABLE backslash_test

┌─statement────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ CREATE TABLE default.backslash_test ( status Enum8(\'VALID\' = 1, \'DELETED\' = 2) DEFAULT CAST(\'VALID\' AS Enum8(\'VALID\' = 1, \'DELETED\' = 2))) ENGINE = Memory │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘

'-- you can't just copy that statement and use that, you need to either change output format to TSVRaw either unbackslash that manually

:) SELECT arrayStringConcat(['\'', '\'', '\''], '\\')

┌─arrayStringConcat([\'\\\'\', \'\\\'\', \'\\\'\'], \'\\\\\')─┐
│ \'\\\'\\\'                                                  │
└─────────────────────────────────────────────────────────────┘
```

I can understand that that backslashing of special characters is a requirement for machine-readable formats like TSV / CSV / Json etc. But why is it also used in human readable formats?

For example (as a contrast) in MySQL:

```sql
-- header formatting sample
mysql> select concat('a','b','c');
+---------------------+
| concat('a','b','c') |
+---------------------+
| abc                 |
+---------------------+
1 row in set (0.01 sec)

mysql> select concat_ws('\n','a','b','c') as x, 1 as y;
+-------+---+
| x     | y |
+-------+---+
| a
b
c | 1 |
+-------+---+
1 row in set (0.00 sec)

mysql> select concat_ws('\n','a','b','c') as x, 1 as y\G
*************************** 1. row ***************************
x: a
b
c
y: 1
1 row in set (0.00 sec)

mysql> select concat_ws('\t','a','b','c') as x, 1 as y\G
*************************** 1. row ***************************
x: a	b	c
y: 1
1 row in set (0.00 sec)

mysql> select concat_ws('\t','a','b','c') as x, 1 as y;
+-------+---+
| x     | y |
+-------+---+
| a	b	c | 1 |
+-------+---+
1 row in set (0.00 sec)
```

Postgres psql:
```sql
-- header formatting sample
postgres=# SELECT concat('a', 'b','c');
 concat 
--------
 abc
(1 row)

postgres=# SELECT concat_ws(E'\n', 'a', 'b','c') as x, 1 as y;
 x | y 
---+---
 a+| 1
 b+| 
 c | 
(1 row)

postgres=# \x
Expanded display is on.
postgres=# SELECT concat_ws(E'\n', 'a', 'b','c') as x, 1 as y;
-[ RECORD 1 ]
x | a
  | b
  | c
y | 1

postgres=# SELECT concat_ws(E'\t', 'a', 'b','c') as x, 1 as y;
-[ RECORD 1 ]--------
x | a       b       c
y | 1

postgres=# \x
Expanded display is off.
postgres=# SELECT concat_ws(E'\t', 'a', 'b','c') as x, 1 as y;
         x         | y 
-------------------+---
 a       b       c | 1
(1 row)

```

**Comment by alexey-milovidov** [comment]
The behaviour is motivated by following:
- don't break table formatting in most cases (I see that PostgreSQL do it better);
- use the same method as in TSV of for contents of quoted strings.

> -- it looks like TSVRaw is the only printable format which doesn't do that

Also VerticalRaw.

Currently I'm not sure that we should change this behaviour. We should think further...

For column names we are going to automatically add aliases like `_1`, `_2` for complex expressions in resultset.

**Comment by filimonov** [comment]
About Pretty* format - it's not so clear how to deal with that. One of the possible options is just "document & let user choose".

For now that behavior leads to some misunderstanding what's in DB, especially if you're inserting some strings with slashes or special characters, and trying to check how the data look like in DB.

> Also VerticalRaw

That is not documented, but actually that's what I've needed. Probably should be added to docs.

Issue can be closed.

**Comment by alexey-milovidov** [comment]
We decided that we don't need to apply any escaping in Pretty formats.

**Comment by alexey-milovidov** [comment]
The only issue is that you wouldn't be able to differentiate between NULL and string with 'NULL' value.
I'm still in doubt.

**Comment by alexey-milovidov** [comment]
Released in 1.1.54388. See changelog.
